### PR TITLE
Implement `BITS` constant for non-zero integers

### PR DESCRIPTION
**Pull request by nvzqz**
This adds the associated [`BITS`](https://doc.rust-lang.org/stable/std/primitive.usize.html#associatedconstant.BITS) constant to `NonZero{U,I}{8,16,32,64,128,size}`.

This is useful when a type alias refers to either a regular or non-zero integer.

### CODE REVIEW

File: library/core/src/num/nonzero.rs [hunk 1]
```diff
@@ -988,3 +988,52 @@ macro_rules! nonzero_unsigned_is_power_of_two {
...
+                ///
+                /// ```
+                /// #![feature(nonzero_bits)]
+                #[doc = concat!("# use std::num::", stringify!($Ty), ";")]
+                ///
+                #[doc = concat!("assert_eq!(", stringify!($Ty), "::BITS, ", stringify!($BITS), ");")]
+                #[doc = concat!("assert_eq!(", stringify!($Ty), "::BITS, ", stringify!($Int), "::BITS);")]
+                /// ```
+                #[unstable(feature = "nonzero_bits", issue = "none")]
+                pub const BITS: u32 = $BITS;
```
Comment (paolobarbolini) [inline library/core/src/num/nonzero.rs:1010]: Drive-by review: could this just be `$Int::BITS` instead of `$BITS`?

Comment (nvzqz) [reply library/core/src/num/nonzero.rs:1010]: I considered implementing it directly in terms of `$Int::BITS` but wanted the constant in doctests. After thinking about it for a bit (heh), I'm in favor of this. Especially because it reduces the maintenance burden.

**Comment by yaahc** [comment]
Looks good, just needs a tracking issue.

**Comment by scottmcm** [comment]
minor thing, since you need another commit for a tracking issue anyway:

Consider using this constant on line 468 in the `log2` implementation.  Doesn't make a big difference, but avoiding some `<$Int>` would be slightly nicer, I think.

**Comment by JohnCSimon** [comment]
Ping from triage: 
@nvzqz can you address the comment from the reviewer?

FYI: when a PR is ready for review, post a message containing 
`@rustbot ready` to switch the PR to `S-waiting-on-review` so the PR appears in the reviewer's backlog.

**Review by dtolnay — APPROVED** [review]
